pkg/restapi/api_test.go

```diff
@@ -17,115 +15,166 @@ import (
 	"github.com/telepresenceio/telepresence/v2/pkg/restapi"
 )
 
-type yesNo bool
+type yesNoClient bool
+type yesNoCluster bool
 
-func (yn yesNo) Intercepts(_ context.Context, _ string, _ http.Header) (bool, error) {
-	return bool(yn), nil
+func (yn yesNoClient) InterceptInfo(_ context.Context, _ string, _ http.Header) (*restapi.InterceptInfo, error) {
+	return &restapi.InterceptInfo{Intercepted: bool(yn), ClientSide: true}, nil
+}
+
+func (yn yesNoCluster) InterceptInfo(_ context.Context, _ string, _ http.Header) (*restapi.InterceptInfo, error) {
+	return &restapi.InterceptInfo{Intercepted: bool(yn), ClientSide: false}, nil
 }
 
 type textMatcher map[string]string
+type textMatcherClient textMatcher
+type textMatcherCluster textMatcher
 
-func (t textMatcher) Intercepts(_ context.Context, _ string, header http.Header) (bool, error) {
+func (t textMatcher) intercepted(header http.Header) bool {
 	for k, v := range t {
 		if header.Get(k) != v {
-			return false, nil
+			return false
 		}
 	}
-	return true, nil
+	return true
+}
+
+func (t textMatcherClient) InterceptInfo(_ context.Context, _ string, header http.Header) (*restapi.InterceptInfo, error) {
+	return &restapi.InterceptInfo{Intercepted: textMatcher(t).intercepted(header), ClientSide: true}, nil
+}
+
+func (t textMatcherCluster) InterceptInfo(_ context.Context, _ string, header http.Header) (*restapi.InterceptInfo, error) {
+	return &restapi.InterceptInfo{Intercepted: textMatcher(t).intercepted(header), ClientSide: false}, nil
+}
+
+type matcherWithMetadata struct {
+	textMatcherCluster
+	meta map[string]string
+}
+
+func (t *matcherWithMetadata) InterceptInfo(c context.Context, p string, header http.Header) (*restapi.InterceptInfo, error) {
+	ret, _ := t.textMatcherCluster.InterceptInfo(c, p, header)
+	ret.Metadata = t.meta
+	return ret, nil
 }
 
 type callerIdMatcher string
```

Comment:
might be worth updating this to `callerIdMatcherCluster` to match the client-cluster pairing that has been added in this commit